src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { PersonPage } from '../PersonPage/PersonPage';
+
+type Props = {
+  people: Person[],
+  setPeople: (newValue: Person[]) => void,
+};
+
+export const PeoplePage: React.FC<Props> = ({ setPeople, people }) => {
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    setHasError(false);
+    setLoading(true);
+
+    (async () => {
+      try {
+        setPeople(await getPeople());
+      } catch {
+        setHasError(true);
+      } finally {
+        setLoading(false);
+      }
+    })();
+
+    return () => setPeople([]);
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {loading && !hasError && !people.length && <Loader />}
+
+          {!loading && hasError && !people.length && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {!loading && !hasError && !people.length && (
+            <p data-cy="noPeopleMessage">
+              There are no people on the server
+            </p>
+          )}
+
+          {!loading && !hasError && people.length
+            && <PersonPage people={people} />}
```

Comment:
Better to create a separate variable for each check

[mateacademy.slack.com/archives/C05DDDBKKV2]( https://mateacademy.slack.com/archives/C05DDDBKKV2/p1695924207111579)

```jsx
  const isDisplayErrorMessage = isError && !isLoading;

  const isNoPeopleOnServer = !people.length && !isLoading && !isError;

  const isPeopleOnServer = !!people.length && !isError;

return (
          {isLoading && (
            <Loader />
          )}

          {isDisplayErrorMessage && (
            <ErrorComponent />
          )}

          {isNoPeopleOnServer && (
            <EmptyComponent />
          )}

          {isPeopleOnServer && (
            <PeopleComponent />
          )}
);
```